pkg/utils/managedresources/registry.go

```diff
@@ -174,7 +211,7 @@ func (r *Registry) objectName(obj client.Object) (string, error) {
 
 	return fmt.Sprintf(
 		"%s__%s__%s",
-		strings.ToLower(gvk.Kind),
+		strings.ToLower(gvk.String()),
```

Comment:
We avoid collisions by using GVK instead of only Kind, for instance when you have the same kind in different groups. This is for internal handling only and backwards compatible.

Reply:
Thanks for the explanation.